File: run_length_encoding/run_length_encoding.java
public class RunLengthEncoding {

  public static String runLengthEncoding(String str) {
    int currentRunLength = 1;
    StringBuilder encodedString = new StringBuilder();

    for (int i = 1; i < str.length(); i++) {
      char previousCharacter = str.charAt(i - 1);
      char currentCharacter = str.charAt(i);

      if (currentCharacter != previousCharacter || currentRunLength == 9) {
        encodedString.append(currentRunLength).append(previousCharacter);
        currentRunLength = 0;
      }

      currentRunLength++;
    }
    
    encodedString.append(currentRunLength).append(str.charAt(str.length() - 1));
    return encodedString.toString();
  }

  public static void main(String[] args) {
      String input = "aaaaaaaaaaaaaaaaabbbbbbbbbbbbbccccccccccc";
      System.out.println(runLengthEncoding(input));
  }
}

File: run_length_encoding/run_length_encoding.ts
function runLengthEncoding(str: string): string {
    let currentRunLength = 1;
    let encodedString: string[] = [];

    for (let i = 1; i < str.length; i++) {
        const previousCharacter = str[i - 1];
        const currentCharacter = str[i];

        if (currentCharacter !== previousCharacter || currentRunLength === 9) {
            encodedString.push(currentRunLength.toString());
            encodedString.push(previousCharacter);
            currentRunLength = 0;
        }

        currentRunLength++;
    }

    encodedString.push(currentRunLength.toString());
    encodedString.push(str[str.length - 1]);

    return encodedString.join('');
}

console.log(runLengthEncoding("aaaaaaaaaaaaaaaaabbbbbbbbbbbbbccccccccccc"));
